app/routes.py

```diff
@@ -1,2 +1,220 @@
+
+    new_task = Task(title=request_data["title"], description=request_data["description"],
+                completed_at=request_data["completed_at"])
+    
+    db.session.add(new_task)
+    db.session.commit()
+
+    return jsonify({"task": new_task.to_dict()}), 201
+    
+
+@tasks_bp.route("", methods=["GET"])
+def list_all_tasks():
+    tasks_response = []
+    #Can't be jsonified
+    if request.args.get('sort') == 'asc':
+        tasks = Task.query.order_by(Task.title.asc()).all()
+
+    elif request.args.get('sort') == 'desc':
+        tasks = Task.query.order_by(Task.title.desc()).all()
+
+    else:
+        tasks = Task.query.all()
+
+    for task in tasks:
+        tasks_response.append(task.to_dict())
+    return jsonify(tasks_response), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT"])
+def handle_task(task_id):
+    print(f"handle task {task_id}")
+    print(f"Request Method {request.method}")
+    task_id = int(task_id)
+    task = Task.query.get(task_id)
+    if task is None:
+        return make_response(f"Task {task_id} not found", 404)
+
+    if request.method == "GET":
+        return jsonify({"task": task.to_dict()}), 200
+        
+    elif request.method == "PUT":
+        input_data = request.get_json()
+        input_data = sanitize_data(input_data)
+        task.title = input_data["title"]
+        task.description = input_data["description"]
+        db.session.commit()
+
+        return jsonify({"task": task.to_dict()}), 200
+
+def sanitize_data(input_data):
+    data_types = {"title": str, "description": str}
+    for name, val_type in data_types.items():
+        try:
+            val = input_data[name]
+            type_test = val_type(val)
+        except Exception as e:
+            abort(400, "Bad Data")
+    return input_data
+
+
+@tasks_bp.route("/<task_id>", methods=["DELETE"])
+def delete_task(task_id):
+    print(f"Delete{task_id}")
+    try:
+        task_id = int(task_id)
+    except ValueError:
+        return 404
+    task = Task.query.get(task_id)
+
+    if task:
+        db.session.delete(task)
+        db.session.commit()
+        response_string = f'Task {task_id} "{task.title}" successfully deleted'
+
+        return jsonify({
+            "details": response_string
+            }), 200
+
+    else:
+        return make_response("", 404)
+
+@tasks_bp.route("/<task_id>/<completion_status>", methods=["PATCH"])
+def mark_complete(task_id, completion_status):
+    
+    task = Task.query.get(task_id)
+    PATH = "https://slack.com/api/chat.postMessage"
+
+    if task is None:
+        return make_response("", 404)
+
+    elif completion_status == "mark_complete":
+        task.completed_at = datetime.utcnow
+        token=os.environ.get("TOKEN")
+        query_params = {
+        "token": token,
+        "channel": "task-notifications",
+        "text": f"Someone just completed task {task.title}"
+    }
+
+        requests.post(PATH, data=query_params)
```

Comment:
It would be nice to abstract out the Slack integration into a separate function.